spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8ConfigMapPropertySource.java

```diff
@@ -49,34 +48,21 @@ public Fabric8ConfigMapPropertySource(KubernetesClient client, String name) {
 
 	public Fabric8ConfigMapPropertySource(KubernetesClient client, String applicationName, String namespace,
 			Environment environment) {
-		super(getName(applicationName, getNamespace(client, namespace)),
-				getData(client, applicationName, getNamespace(client, namespace), environment));
+		super(getName(applicationName, getApplicationNamespace(client, namespace)),
+				getData(client, applicationName, getApplicationNamespace(client, namespace), environment));
 	}
 
 	private static Map<String, Object> getData(KubernetesClient client, String applicationName, String namespace,
 			Environment environment) {
 		try {
-			Map<String, Object> result = new HashMap<>();
-			ConfigMap map = !StringUtils.hasLength(namespace) ? client.configMaps().withName(applicationName).get()
```

Comment:
this code here (and the one down that I have removed), where doing the same exact thing:

- get a `ConfigMap`
- if present, get it's data

I've moved it `Fabric8ConfigUtils::getConfigMapData`